src/mvs/mvsNodeActions.ts

```diff
@@ -68,7 +68,10 @@ export function getDatasetLabel(node: ZoweDatasetNode) {
 export async function uploadFile(node: ZoweDatasetNode, doc: vscode.TextDocument) {
     try {
         const datasetName = getDatasetLabel(node);
-        await ZoweExplorerApiRegister.getMvsApi(node.getProfile()).putContents(doc.fileName, datasetName);
+        const prof = node.getProfile();
+        await ZoweExplorerApiRegister.getMvsApi(prof).putContents(doc.fileName, datasetName, {
+            encoding: prof.profile.encoding
+        });
```

Comment:
This makes me think that instead of having to forward all properties from the user profile, we could just combine the "already constructed" profile (i.e. whenever we take into account the use of base-profile and furthermore respect the order-of-precedence `Env -> Service Profile -> Base Profile` in the APIs on NPM) with whatever options are passed in these functions.
Just a thought. I sure hope it makes some sense 😋

Reply:
It would be great if whenever we added the encoding feature into Zowe CLI, that we only needed to update the version of Zowe Explorer for it to immediately make use of it.  

If that's what you mean, I'm onboard 😄

Reply:
Yeah, Thanks for putting it in fewer words 😋